modules/rubiconBidAdapter.js

```diff
@@ -949,6 +882,63 @@ function addVideoParameters(data, bidRequest) {
   data.imp[0].video.h = size[1]
 }
 
+function applyFPD(fpd, mediaType, data) {
+  const map = {user: {banner: 'tg_v.', code: 'user'}, context: {banner: 'tg_i.', code: 'site'}, adserver: 'dfp_ad_unit_code'};
+  let obj = {};
+  let keywords = [];
+  const validate = function(e, t) {
+    if (typeof e === 'object' && !Array.isArray(e)) {
+      utils.logWarn('Rubicon: Filtered FPD key: ', t, ': Expected value to be string, integer, or an array of strings/ints');
+    } else if (e) {
+      return (Array.isArray(e)) ? e.filter(value => {
+        if (typeof value !== 'object' && value) return value;
+
+        utils.logWarn('Rubicon: Filtered value: ', value, 'for key', t, ': Expected value to be string, integer, or an array of strings/ints');
+      }).toString() : e.toString();
+    }
+  };
+
+  Object.keys(fpd).filter(value => fpd[value] && map[value] && typeof fpd[value] === 'object').forEach((type) => {
+    obj[map[type].code] = Object.keys(fpd[type]).filter(value => fpd[type][value]).reduce((result, key) => {
+      if (key === 'keywords') {
+        if (!Array.isArray(fpd[type][key]) && mediaType === BANNER) fpd[type][key] = [fpd[type][key]]
+
+        result[key] = fpd[type][key];
+
+        if (mediaType === BANNER) keywords = keywords.concat(fpd[type][key]);
```

Comment:
The [setBidderConfig docs ](https://docs.prebid.org/features/firstPartyData.html) have keywords value as a comma separated string. When passing the data this way, the merge for **keywords** between fpd.context.keywords, fpd.user.keywords, and params.keywords the **kw** property will be missing the params.keywords value.

Reply:
Yeah, so we should add logic to convert our bidParams.keywords into a comma separated string.

if (bidRequest.params.keywords) bidFpd.context.keywords = bidRequest.params.keywords.join(',');

Reply:
I _think_ this would solve it

Reply:
Corrected this. Note this is setup as an overwrite not a merge. Priority as follows:
1. params.keywords
2. bidRequest.keywords
3. config.fpd.keywords

Reply:
Could you add a docs PR with the above noted on your bidders page

Reply:
https://github.com/prebid/prebid.github.io/pull/2660